common/src/main/java/io/netty/util/internal/ReferenceCountUpdater.java

```diff
@@ -80,8 +65,7 @@ public final int refCnt(T instance) {
 
     public final boolean isLiveNonVolatile(T instance) {
         final int rawCnt = getRawRefCnt(instance);
-        // The "real" ref count is > 0 if the rawCnt is even.
-        return rawCnt == 2 || rawCnt == 4 || rawCnt == 6 || rawCnt == 8 || (rawCnt & 1) == 0;
+        return (rawCnt & 1) == 0;
```

Comment:
```suggestion
        if (rawCnt == 2) {
            return true;
        }    
        return (rawCnt & 1) == 0;
```
This was a decent fast path to save slowing down making the branch taken non-speculative again and move the pipeline forward